testplan/common/entity/base.py

```diff
@@ -141,10 +139,11 @@ def start(self):
                 resource.start()
                 if not resource.cfg.async_start:
                     resource.wait(resource.STATUS.STARTED)
-            except Exception as exc:
-                msg = 'While starting resource [{}]{}{}'.format(
-                    resource.cfg.name, os.linesep,
-                    format_trace(inspect.trace(), exc))
+            except Exception:
+                msg = 'While starting resource [{}]\n{}'.format(
```

Comment:
`\n`  is compatible with Windows and Linux on the console. But It's not compatible on the log files. I think `os.linesep` is a good multiplatform solution.

Reply:
If a file is opened in text mode (the default), python will convert any \n to \r\n before writing:

```
Python 3.7.1 (v3.7.1:260ec2c36a, Oct 20 2018, 14:57:15) [MSC v.1915 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> with open("file.txt", "w") as f:
...     f.write('\n'.join(["line", "line", "yet another line"]))
...
26
Python 3.7.1 (v3.7.1:260ec2c36a, Oct 20 2018, 14:57:15) [MSC v.1915 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> with open("file.txt", "rb") as f:
...     byte_contents = f.read()
...
>>> print(byte_contents)
b'line\r\nline\r\nyet another line'
```

See https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files

> When writing in text mode, the default is to convert occurrences of \n back to platform-specific line endings.

Reply:
The _only_ reason to use `os.linesep` is if you are writing to a file or socket opened in _binary_ mode. I'm not aware of anywhere in Testplan in which we do that.